pkg/scalers/scaler.go

```diff
@@ -168,29 +168,74 @@ func GenerateMetricInMili(metricName string, value float64) external_metrics.Ext
 	}
 }
 
+type Option func(*configOptions)
+
+type configOptions struct {
+	useMetadata       bool
+	useAuthentication bool
+	useResolvedEnv    bool
+	isOptional        bool
+	defaultVal        interface{}
+}
+
+func UseMetadata(metadata bool) Option {
+	return func(opt *configOptions) {
+		opt.useMetadata = metadata
+	}
+}
+
+func UseAuthentication(auth bool) Option {
+	return func(opt *configOptions) {
+		opt.useAuthentication = auth
+	}
+}
+
+func UseResolvedEnv(resolvedEnv bool) Option {
+	return func(opt *configOptions) {
+		opt.useResolvedEnv = resolvedEnv
+	}
+}
+
+func IsOptional(optional bool) Option {
+	return func(opt *configOptions) {
+		opt.isOptional = optional
+	}
+}
+
+func WithDefaultVal(defaultVal interface{}) Option {
+	return func(opt *configOptions) {
+		opt.defaultVal = defaultVal
+	}
+}
+
 // getParameterFromConfigV2 returns the value of the parameter from the config
-func getParameterFromConfigV2(config *scalersconfig.ScalerConfig, parameter string, useMetadata bool, useAuthentication bool, useResolvedEnv bool, isOptional bool, defaultVal string, targetType reflect.Type) (interface{}, error) {
+func getParameterFromConfigV2(config *scalersconfig.ScalerConfig, parameter string, targetType reflect.Type, options ...Option) (interface{}, error) {
```

Comment:
please extend the docs for this function a bit. So newcomers understand what should be the input and what it does under the hood.